### PR TITLE
Add stop-and-copy script for database migration

### PR DESCRIPTION
**Pull request by sldblog**
We are migrating to PostgreSQL 14 by setting up a completely new instance.

To facilitate the migration, we need to:

1. Gracefully stop serving traffic
2. Ensure everything has stopped
3. Copy the contents of the PostgreSQL 10 database to PostgreSQL 14
4. Re-deploy the application with the new configuration pointing to the new database

This commit adds a script doing the first 3 points on dev.

### CODE REVIEW

**Comment by sldblog** [comment]
[Update postgres to latest CP supported version as 10 will be dead in November 2022](https://trello.com/c/vge330Xk/320-update-postgres-to-latest-cp-supported-version-as-10-will-be-dead-in-november-2022)

File: copy_dev_postgres_to_v14.sh [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+sleep 2
+
+
+echo
+echo "✨ Stopping UI"
+kubectl scale \
+  --namespace="$target_namespace" \
+  --replicas=0 \
+  --timeout=5m \
+  deployment/hmpps-interventions-ui
```
Comment (sldblog) [inline copy_dev_postgres_to_v14.sh:15]: "Scale to 0 replicas" means no instance of the UI application will be running, effectively cutting off the service from end users.

All in-progress requests can complete within the grace period (10 seconds), which is usually more than enough for all requests to complete.

File: copy_dev_postgres_to_v14.sh [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  pod
+
+
+echo
+echo "✨ Stopping API/jobs"
+kubectl scale \
+  --namespace="$target_namespace" \
+  --replicas=0 \
+  --timeout=5m \
+  --selector=app.kubernetes.io/name=hmpps-interventions-service \
```
Comment (sldblog) [inline copy_dev_postgres_to_v14.sh:32]: "Scale to 0" on the service/API/batch jobs level, ensuring we don't have an active database connection.

Note: this won't prevent scheduled jobs from kicking off.

File: copy_dev_postgres_to_v14.sh [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  --selector=release=hmpps-interventions-service \
+  pod
+
+
+echo
+echo "✨ Starting copy job"
+
+kubectl apply \
+  --namespace="$target_namespace" \
+  --filename=./copy-from-postgres10-to-postgres14.yaml
```
Comment (sldblog) [inline copy_dev_postgres_to_v14.sh:49]: Executes pg_dump/pg_restore between the databases.

File: copy_dev_postgres_to_v14.sh [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+echo
+echo "✨ Waiting for copy job to finish"
+
+kubectl wait --for=condition=Complete --timeout=-1s \
+  --namespace="$target_namespace" \
+  job/db-copy-once
```
Comment (sldblog) [inline copy_dev_postgres_to_v14.sh:56]: Waits for the copy job to finish.

**Review by kth13 — APPROVED** [review]
LGTM

**Pull request merged**